lua/luasnip/loaders/from_snipmate.lua

```diff
@@ -0,0 +1,218 @@
+---@diagnostic disable: redefined-local
+local ls = require("luasnip")
+local caches = require("luasnip.loaders._caches")
+local util = require("luasnip.util.util")
+local Path = require("luasnip.util.path")
+local session = require("luasnip.session")
+
+local function parse_snipmate(buffer, filename)
+	local snippet = {}
+	local extends = {}
+
+	---@type string[]
+	local lines = vim.split(buffer, "\n")
+	local i = 1
+
+	local function _parse()
+		local line = lines[i]
+		local prefix = line:match("%s+(%S+)")
+		local description = line:match([["(.+)"%s*$]])
```

Comment:
What do you think about replacing this with a single match like `^snippet%s+(%S+)%s+"(.*)"%s*$`, IMO that's a bit clearer

Reply:
You are right. There may be no explanation, so `^snippet%s+(%S+)%s*"(. *)"%s*$`, I guess.

Reply:
Oh, that's not gonna work.

Reply:
```lua
local prefix = line:match([[^snippet%s+(%S+)]])
local description = line:match([[^snippet%s+$S+$s+"(.*)"%s*$]])
```
If you don't separate them, you can't get a prefix when there is no description.

Reply:
Ah, because of the `"`. We could make them optional with `"?` (btw, are you sure they are needed? From what I can tell all text behind the trigger should be interpreted as dscr)

Reply:
I've never actually used snipmate HAHAHA. Certainly not necessary.

Reply:
xD
Okay, that makes it simpler :D

Reply:
```lua
local prefix, description = line:match([[^snippet%s+(%S+)%s*(.*)]])
```
This looks good.